src/fu-install-task.c

```diff
@@ -92,6 +92,59 @@ fu_install_task_get_is_downgrade (FuInstallTask *self)
 	return self->is_downgrade;
 }
 
+static gboolean
+fu_install_task_check_verfmt (FuInstallTask *self,
+			      GPtrArray *verfmts,
+			      FwupdInstallFlags flags,
+			      GError **error)
+{
+	FwupdVersionFormat fmt_dev = fu_device_get_version_format (self->device);
+	g_autoptr(GString) version_format = g_string_new (NULL);
+
+	/* for the error message */
+	for (guint i = 0; i < verfmts->len; i++) {
+		XbNode *verfmt = g_ptr_array_index (verfmts, i);
+		const gchar *tmp = xb_node_get_text (verfmt);
+		g_string_append_printf (version_format, "%s;", tmp);
+	}
+	if (version_format->len > 0)
+		g_string_truncate (version_format, version_format->len - 1);
+
+	/* no device format */
+	if (fmt_dev == FWUPD_VERSION_FORMAT_UNKNOWN &&
+	    (flags & FWUPD_INSTALL_FLAG_FORCE) == 0) {
+		g_set_error (error,
+			     FWUPD_ERROR,
+			     FWUPD_ERROR_NOT_SUPPORTED,
+			     "release version format '%s' but no device version format",
+			     version_format->str);
+		return FALSE;
+	}
+
+	/* fall back to older version formats */
```

Comment:
this text is wrong; it's not falling back - it's just checking all version formats.

Reply:
Fixed.